src/Common/src/System/Security/Cryptography/CngPkcs8.cs

```diff
@@ -0,0 +1,336 @@
+            byte[] rented = ArrayPool<byte>.Shared.Rent(key.KeySize);
+            int rentWritten = 0;
+
+            // If we use 6 bits from each byte, that's 22 * 6 = 132
+            Span<char> randomString = stackalloc char[22];
+
+            try
+            {
+                FillRandomAsciiString(randomString);
+
+                while (!key.TryExportEncryptedPkcs8PrivateKey(
+                    randomString,
+                    s_platformParameters,
+                    rented,
+                    out rentWritten))
+                {
+                    int size = rented.Length;
+                    ArrayPool<byte>.Shared.Return(rented);
+                    rented = ArrayPool<byte>.Shared.Rent(checked(size * 2));
+                }
+
+                return KeyFormatHelper.ReencryptPkcs8(
+                    randomString,
+                    rented.AsMemory(0, rentWritten),
+                    passwordBytes,
+                    pbeParameters);
+            }
+            finally
+            {
+                randomString.Clear();
+                CryptographicOperations.ZeroMemory(rented.AsSpan(0, rentWritten));
+                ArrayPool<byte>.Shared.Return(rented);
+            }
+        }
+
+        private static AsnWriter RewriteEncryptedPkcs8PrivateKey(
+            AsymmetricAlgorithm key,
+            ReadOnlySpan<char> password,
+            PbeParameters pbeParameters)
+        {
+            Debug.Assert(pbeParameters != null);
+
+            byte[] rented = ArrayPool<byte>.Shared.Rent(key.KeySize);
+            int rentWritten = 0;
+
+            try
+            {
+                while (!key.TryExportEncryptedPkcs8PrivateKey(
+                    password,
+                    s_platformParameters,
+                    rented,
+                    out rentWritten))
+                {
+                    int size = rented.Length;
+                    ArrayPool<byte>.Shared.Return(rented);
+                    rented = ArrayPool<byte>.Shared.Rent(checked(size * 2));
+                }
+
+                return KeyFormatHelper.ReencryptPkcs8(
+                    password,
+                    rented.AsMemory(0, rentWritten),
+                    password,
+                    pbeParameters);
+            }
+            finally
+            {
+                CryptographicOperations.ZeroMemory(rented.AsSpan(0, rentWritten));
+                ArrayPool<byte>.Shared.Return(rented);
+            }
+        }
+
+        private static void FillRandomAsciiString(Span<char> destination)
+        {
+            Debug.Assert(destination.Length < 128);
+            Span<byte> randomKey = stackalloc byte[destination.Length];
+            RandomNumberGenerator.Fill(randomKey);
+
+            for (int i = 0; i < randomKey.Length; i++)
+            {
+                // 33 (!) up to 33 + 63 = 96 (`)
+                destination[i] = (char)(33 + (randomKey[i] & 0b0011_1111));
+            }
```

Comment:
Could you just fill the original destination and then overwrite it? e.g. pseudo-code:
```C#
RandomNumberGenerator.Fill(MemoryMarshal.AsBytes(destination));
for (int i = 0; i < destination.Length; i++)
{
    destination[i] = (char)(33 + (destination[i] & 0b0011_1111));
}
```
?